enhancements/sig-architecture/47-manifestwork-updatestrategy/README.md

```diff
@@ -0,0 +1,223 @@
+will update deployment constantly.
+
+#### Apply failure
+
+Some field of the resources has both defaulting and immutable set, with which work agent is unable to update the resource. `Job` is an example. If user does not set `spec.Selector` of job, the kube-apiserver will set a default selector and label on pod. 
+These fields are immutable, so work agent will fail later when it compares the job manifest and tries to update the job.
+
+#### Fight with other actors
+
+A resource might not only managed by work agent, but also other actors. Other actors would want to mutate some field in the resource
+which will later be rolled back by the work agent. Some use cases may need these actors to control some fields of the resources other than work agent.
+
+A possible solution to resolve the first two problems is that work-agent records the hash of required spec and the generation of the last applied resource, and instead of comparing required to the existing resources, work-agent compares hash (if required changes) and generation (if existing changes). However, it cannot resolve the third problem involving multiple actors.
+
+## Motivation
+
+Provide an interface for user to control how resource defined in `ManifestWork` should be updated.
+
+## Proposal
+
+We propose to update `ManifestWork` API so the user can specify upstrategy for each resource manifest in the `ManifestWork`. Possible strategy includes:
+
+1. by default, work-agent update resource as what is implemented today.
+2. user can choose not to update the resource, work-agent will only check the existence of the resource, but not keep its spec update.
+3. user can choose to use server side apply when work-agent update the resource.
+4. to support lowner version of kube, we may also cosider allowing user to choose json merge approach to update the resource.
+
+### Design Details
+
+#### API change
+
+We could add an updateStraty for each manifest as below:
+
+```go
+type UpdateStrategy struct {
+	// type defines the strategy to update this manifest, default value is Update.
+	// Update type means to update resource by an update call.
+	// None type means do not update resource based on current manifest.
+	// ServerSideApply type means to update resource using server side apply as fieldManager of work-controller.
+	// If there is conflict, the related Applied confition of manifest will in the status of False with the
+	// reason of ApplyConflict.
+	// +kubebuilder:default=60
+	// +kubebuilder:validation:Enum=Update;None;ServerSideApply
+	// +kubebuilder:validation:Required
+	// +required
+	Type string `json:"type"`
+
+	// serverSideApply defines the configuration for server side apply. It is honored only when
+	// type of updateStrategy is ServerSideApply
+	// +optional
+	ServerSideApply *ServerSideApplyConfig `json:"serverSideApply,omitempty"`
+}
+
+type ServerSideApplyConfig struct {
+	// Force represents to force apply the manifest.
+	// +optional
+	Force bool `json:"force"`
```

Comment:
do we need to clarify the field-manager here?

Reply:
and are users expected to flip the "force" bit to false after the force apply worked for one time?

Reply:
the field-manager will be `work-controller` or `work-agent`, I do not think of a use case that use needs to set another field manager yet. Thought?

Reply:
>  I do not think of a use case that use needs to set another field manager yet.

agree, i just think it would be better if we can reflect the default value in the api object